plugins/catalog-backend/src/ingestion/processors/BuiltinKindsEntityProcessor.ts

```diff
@@ -233,6 +282,29 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
         RELATION_MEMBER_OF,
         RELATION_HAS_MEMBER,
       );
+
+      // TODO: Create an attachment for the photo by downloading the URL.
+
+      // TODO: Property name could be better... like pictureLocation...
+
+      if (user.spec.profile?.picture) {
+        const response = await fetch(user.spec.profile?.picture);
```

Comment:
Should still use the UrlReader because auth and stuff.

I guess we'll want to handle errors as warnings for non-essential data eventually.

Reply:
Yes, in the final solution this should use the URL reader interface. Right now the problem is, that `UrlReader.read()` returns a `Buffer`. However, I need access to the content type, so we have to change the interface. We should return an object with a buffer and content type instead. However, this would be a breaking change and requires some more work. This is one reason why I would like to move the actual usage of the attachments in a follow up PR.

An alternative would be guessing the mime type, not sure if that is a good idea.

Reply:
There's a bit of a security concern here too where you can access arbitrary endpoints and their data if you control the user entity definition, which is very really not great. Perhaps it's better to use the URL reader for that reason?

Guessing mime type is probably gonna be fine, afaik most image formats have some nice bytes set to make that easy, thinking we just slap https://www.npmjs.com/package/image-type on it.

+1 to refactoring UrlReader though, idea would be that we deprecate `read` and introduce `readUrl` or something like that, which then returns a response object similar to `readTree` etc.

Reply:
> +1 to refactoring UrlReader though, idea would be that we deprecate read and introduce readUrl or something like that, which then returns a response object similar to readTree etc.

But that is a task for a separate PR 😉 We could do that before continuing here.